moped-editor/src/views/projects/projectView/ProjectFiles.js

```diff
@@ -357,16 +357,15 @@ const ProjectFiles = ({ handleSnackbar }) => {
   // handles row delete
   const handleDeleteClick = useCallback(
     (id) => () => {
-      // remove row from rows in state
-      setRows(rows.filter((row) => row.project_file_id !== id));
-
       deleteProjectFileAttachment({
         variables: {
           fileId: id,
         },
       })
         .then(() => refetch())
         .then(() => {
+          // remove row from rows in state
+          setRows(rows.filter((row) => row.project_file_id !== id));
```

Comment:
Work and looks great! I noticed that in some places we do:

```js
refetch().then(() => {
// closes snackbar and other things
});
```
What do you think about updating so the UI is frozen until the table data refetch is complete? I'm also open to creating an issue to find a way to iron these out across the board. It is feeling a little like Whac-a-Mole to me but I could be fretting too much.

Reply:
nvm we talked about this in dev sync, and then I proved to myself that this doesn't matter. Thanks for bearing with my over-thinking. Approving! 🚀